cache_test.go

```diff
@@ -432,20 +434,26 @@ func Test_ConcurrentClearAndSet(t *testing.T) {
 		// list is maintained by the background worker. This can create a period
 		// where the two are out of sync. Even SyncUpdate is helpless here, since
 		// it can only sync what's been written to the buffers.
-		for i := 0; i < 10; i++ {
+		assertEventually(t, func() bool {
 			expectedCount := 0
 			if cache.list.Head != nil {
 				expectedCount = 1
 			}
 			actualCount := cache.ItemCount()
-			if expectedCount == actualCount {
-				return
-			}
-			time.Sleep(time.Millisecond)
+			return expectedCount == actualCount
+		}, 100, 50*time.Millisecond, fmt.Sprintf("cache list and lookup are not consistent in iteration %v", i))
```

Comment:
It doesn't matter what numbers I pass here, i always get a failure. @karlseguin i think something is buggy in the code itself, not the test 😨